src/codegen.rs

```diff
@@ -66,20 +66,20 @@ pub trait Generator {
 
     #[inline(never)]
     fn write_string_complex(&mut self, string: &str, mut start: usize) -> io::Result<()> {
-        self.write(string[ .. start].as_bytes())?;
+        self.write(&string.as_bytes()[ .. start])?;
```

Comment:
the changes in this function prevent panics when indexing into a string at a byte offset that doesn't fall on a character boundary